R/PostDefinition.R

```diff
@@ -62,12 +62,34 @@ postDefinition <- function(baseUrl, name, category, definition) {
   } else {
     expression <- definition
   }
-  # convert R-object to JSON expression.
-  jsonExpression <- RJSONIO::toJSON(expression)
-  # create json body
-  json <- paste0("{\"name\":\"", as.character(name), "\",\"expressionType\": \"SIMPLE_EXPRESSION\", \"expression\":",
-                 jsonExpression,
-                 "}")
+  
+  categoryMetaData <- getDefinitionsMetadata(baseUrl = baseUrl, category = category)
+  while(name %in% categoryMetaData$name){
```

Comment:
My concern is cases where a definition is impossible to import without manually editing names in the json definition (this could get messy, especially for definitions with embedded cohort definitions). Overwriting would solve this unless the existing definition is owed by another ATLAS username - in these cases overwriting may not be possible or appropriate.

Would you consider adding an argument to all postDefinition functions that specifies how to deal with name clashes?

Option 1 ('error') could be a meaningful error message. Option 2 ('overwrite') could attempt to overwrite. Option 3 ('rename') could rename with an information message giving details of the renaming.  It would make sense for Option 1 to be the default.

Reply:
Yes, adding such an argument would be useful for sure. Agree with option 1 being default

Reply:
A```duplicateNames``` argument  has been added to ```post%categoryFirstUpper%Definition``` in ```R/templateWebApiFunctions.R``` and hence all post functions in ```R/AutoGeneratedDefinitions.R```. 

The argument can take values ```c("error", "overwrite", "rename")``` and is passed into the ```postDefinition``` function, where it determines the behavior of ```.checkModifyDefinitionName```.